### DISCUSSION TITLE
When calculating the topocentric coordinates of a satellite, how are the coordinate axes defined?

### DISCUSSION THREAD

**Discussion by Rairey233** [skyfielders/python-skyfield#843] [Q&A]
I am trying to convert the topocentric coordinates I get into Unity world space (which is a left handed coordinate system,Y+ is always up, X+ is always right, and Z+ is always forward ).  I calculate the topocentric coordinates with the following:

SAT = EarthSatellite(Line1, Line2)

t = ts.from_astropy(attitude_20210320['time'][i])

LP = wgs84.latlon(28.7603135, -17.8796168, elevation_m=2387)

difference = SAT - LP

topocentric = difference.at(t)

earth = eph['earth']

sun = eph['sun']

E_Sat_vec = topocentric.position.km

However, I wasn't sure on where these x,y,z axes actually are, other than they must form a right handed coordinate system. I would also be interested to know how to transform these coordinates into the Unity world space as there is a lack of documentation on the matter.

**Comment by brandon-rhodes** [comment]
> However, I wasn't sure on where these x,y,z axes actually are, other than they must form a right handed coordinate system. I would also be interested to know how to transform these coordinates into the Unity world space as there is a lack of documentation on the matter.

The axes that Skyfield uses by default for astronomical coordinates are described here, near the top of the "Positions" page of the documentation:

https://rhodesmill.org/skyfield/positions.html#the-icrs-reference-system-and-j2000

I don't know anything about the Unity world space, though, so I can't guess how you might perform a transformation between them. Is there documentation you could point us at about the Unity world space and its orientation?
